test/bats.bats

```diff
@@ -483,3 +483,77 @@ END_OF_ERR_MSG
   # In theory it should take 3s, but let's give it bit of extra time instead.
   [[ "$duration" -lt 20 ]]
 }
+
+@test "randomise test execution order with --shuffle" {
+  # We can't be sure what the order will be -- but it should be different
+  # between different runs. We ignore all #-lines because they always include
+  # BATS_SHUFFLE_SEED which should always be different anyway.
+
```

Comment:
Not sure how it is made sure that the several runs will not accidentally produce the same order. I guess that even different seeds could result in the same order.

If there is a slight chance of a collision (same seed or same order with different seed) i suggest calling with fixed seeds that are known to be free of collisions.